packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -168,20 +190,26 @@ class ValidationResult {
   final List<ValidationMessage> messages;
 
   String get leadingBox {
-    if (type == ValidationType.missing)
-      return '[✗]';
-    else if (type == ValidationType.installed)
-      return '[✓]';
-    else
-      return '[-]';
+    switch(type) {
+      case ValidationType.missing:
+        return '[✗]';
+      case ValidationType.installed:
+        return '[✓]';
+      case ValidationType.partial:
+        return '[!]';
+      default:
+        return '[!]';
+    }
```

Comment:
What we normally do is just return null (and `assert(type != null)` before the switch).
This will get cleaned up once we have non-nullable types.

Reply:
done

Reply:
done